lib/dom_utils.js

```diff
@@ -323,7 +323,7 @@ var DomUtils = {
       // In firefox prior to 96, simulating a click on an element would be blocked. In 96+,
       // extensions can trigger native click listeners on elements. See #3985.
       const mayBeBlocked = event === "click" && Utils.isFirefox() && parseInt(Utils.firefoxVersion()) < 96
-          && /^91\.[0-5](\.|$)/.test(Utils.firefoxVersion())
+          && /^(?!91\.)|^91\.[0-5](\.|$)/.test(Utils.firefoxVersion())
```

Comment:
This regexp is now pretty complicated to look at, and the comment doesn't indicate why we're checking for Firefox 91.

I think this would be much easier to work with if 1) we added firefoxMajorVersion and firefoxMinorVersion to Utils and used those for comparisons, so we don't have to use parseInt on a string and regexps to parse those out here, and 2) updated the comment to describe why we're checking against Firefox 91 in this section.

Reply:
Hello Philc, I've updated the code to make it clear.